### PR TITLE
Refactoring of rxmessenger to allow persistent connections

### PR DESCRIPTION
**Pull request by andreas-aevi**
… and same client id for multiple messages

- Introduced package structure
- ObservableMessengerClient now takes a service ComponentName in the constructor, meaning it's tied to a particular service
- Once bound, the connection will remain open until end of remote stream or closed by client
- Client id is now generated by the client and passed to the service

### CODE REVIEW

File: rxmessenger/src/main/java/com/aevi/android/rxmessenger/MessageConstants.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package com.aevi.android.rxmessenger;
```
Comment (brettwold) [inline rxmessenger/src/main/java/com/aevi/android/rxmessenger/MessageConstants.java]: Copyright - same in many new classes here

Comment (andreas-aevi) [reply rxmessenger/src/main/java/com/aevi/android/rxmessenger/MessageConstants.java]: Ah yes...doh

File: rxmessenger/src/main/java/com/aevi/android/rxmessenger/client/MessengerConnection.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+import android.util.Log;
+
+import java.util.UUID;
+
+import io.reactivex.Observable;
+import io.reactivex.subjects.BehaviorSubject;
+
+import static com.aevi.android.rxmessenger.MessageConstants.*;
+
+class MessengerConnection implements ServiceConnection {
```
Comment (louis-barman) [inline rxmessenger/src/main/java/com/aevi/android/rxmessenger/client/MessengerConnection.java]: Should this be package level? or public???

File: rxmessenger/src/main/java/com/aevi/android/rxmessenger/client/ObservableMessengerClient.java [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+        Log.d(TAG, "Creating client for service: " + serviceComponentName.flattenToShortString());
+        this.context = context;
+        this.serviceComponentName = serviceComponentName;
+        this.onHandleMessageCallback = onHandleMessageCallback;
+    }
+
+    /**
+     * Default handler proxies the message straight to the client.
+     */
+    void handleMessage(String data, String sender, Subject<String> callbackEmitter) {
```
Comment (louis-barman) [inline rxmessenger/src/main/java/com/aevi/android/rxmessenger/client/ObservableMessengerClient.java]: Should this be package level? or public???

**Comment by louis-barman** [comment]
You also need to update the README.md with updated instructions on how to use the new ObservableMessengerClient

File: rxmessenger/src/main/java/com/aevi/android/rxmessenger/client/MessengerConnection.java (hunk 2 above)
Comment (andreas-aevi) [reply rxmessenger/src/main/java/com/aevi/android/rxmessenger/client/MessengerConnection.java]: It should be package level.

File: rxmessenger/src/main/java/com/aevi/android/rxmessenger/client/ObservableMessengerClient.java (hunk 3 above)
Comment (andreas-aevi) [reply rxmessenger/src/main/java/com/aevi/android/rxmessenger/client/ObservableMessengerClient.java]: The things that are package protected are so for a reason.

**Pull request merged**